server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaksBuffer.kt

```diff
@@ -0,0 +1,821 @@
+		rightFootPressure *= 1.0f / rightDistance
+
+		// normalize the pressure values
+		val pressureSum = leftFootPressure + rightFootPressure
+		leftFootPressure /= pressureSum
+		rightFootPressure /= pressureSum
+		return floatArrayOf(leftFootPressure, rightFootPressure)
+	}
+
+	// perform a gradient descent to find the force vectors that best match the
+	// acceleration of the com
+	private fun findForceVectors(leftFootForceInit: Vector3, rightFootForceInit: Vector3) {
+		var leftFootForce: Vector3 = leftFootForceInit
+		var rightFootForce: Vector3 = rightFootForceInit
+		val iterations = 100
+		val stepSize = 0.01f
+		// set up the temporary variables
+		var tempLeftFootForce1: Vector3
+		var tempLeftFootForce2: Vector3
+		var tempRightFootForce1: Vector3
+		var tempRightFootForce2: Vector3
+		var error: Vector3
+		var error1: Vector3
+		var error2: Vector3
+		var error3: Vector3
+		var error4: Vector3
+		for (i in 0 until iterations) {
+			tempLeftFootForce1 = leftFootForce
+			tempLeftFootForce2 = leftFootForce
+			tempRightFootForce1 = rightFootForce
+			tempRightFootForce2 = rightFootForce
+
+			// get the error at the current position
+			error = centerOfMassAcceleration
+				.minus(leftFootForce.plus(rightFootForce).plus(GRAVITY))
+
+			// add and subtract the error to the force vectors
+			tempLeftFootForce1 = tempLeftFootForce1.times(1.0f + stepSize)
+			tempLeftFootForce2 = tempLeftFootForce2.times(1.0f - stepSize)
+			tempRightFootForce1 = tempRightFootForce1.times(1.0f + stepSize)
+			tempRightFootForce2 = tempRightFootForce2.times(1.0f - stepSize)
+
+			// get the error at the new position
+			error1 = getForceVectorError(tempLeftFootForce1, rightFootForce)
+			error2 = getForceVectorError(tempLeftFootForce2, rightFootForce)
+			error3 = getForceVectorError(tempRightFootForce1, leftFootForce)
+			error4 = getForceVectorError(tempRightFootForce2, leftFootForce)
+
+			// set the new force vectors
+			if (error1.len() < error.len()) {
+				leftFootForce = tempLeftFootForce1
+			} else if (error2.len() < error.len()) {
+				leftFootForce = tempLeftFootForce2
+			}
+			if (error3.len() < error.len()) {
+				rightFootForce = tempRightFootForce1
+			} else if (error4.len() < error.len()) {
+				rightFootForce = tempRightFootForce2
+			}
+		}
+	}
+
+	// detect any outside forces on the body such
+	// as a wall or a chair. returns true if there is an outside force
+	private fun detectOutsideForces(f1: Vector3, f2: Vector3): Boolean {
+		val force: Vector3 = GRAVITY.plus(f1).plus(f2)
+		val error: Vector3 = centerOfMassAcceleration.minus(force)
+		return error.lenSq() > FORCE_ERROR_TOLERANCE_SQR
+	}
+
+	// simple error function for the force vector gradient descent
+	private fun getForceVectorError(testForce: Vector3, otherForce: Vector3): Vector3 {
+		return centerOfMassAcceleration
+			.minus(testForce.plus(otherForce).plus(GRAVITY))
```

Comment:
```suggestion
		return centerOfMassAcceleration - testForce + otherForce + GRAVITY
```
Just as an example, you can do this in Kotlin, which makes it easier to read IMO. You can look through your code and use +, -, * and / wherever